RecoLocalCalo/HcalRecAlgos/interface/HcalDeterministicFit.h

```diff
@@ -151,27 +141,22 @@ void HcalDeterministicFit::apply(const CaloSamples & cs, const std::vector<int>
         double invG = invGpar[0]+invGpar[1]*std::sqrt(2*std::log(invGpar[2]/ratio));
         float iG=0;
         getLandauFrac(-invG,-invG+tsWidth,iG);
-        ch4=(corrCharge[4]-ch3*n3)/(iG);
-        ch5=negThresh[0];
-        tsShift4=invG;
+        if (iG != 0 ) {
+	  ch4=(corrCharge[4]-ch3*n3)/(iG);
+	  ch5=negThresh[0];
```

Comment:
@cvuosalo 
As I mentioned this class has been used and will be used in the 80X for trigger for the whole Run2. Hopefully will not even be used at all for the phase1. The working direction is to make the M2 so fast that can even go online. We should have this before the new 2017 GEN-SIM campain  start. So prefer to not touch it anymore with aestethic cleanup or rewriting it.